.travis.yml

```diff
@@ -27,7 +27,9 @@ matrix:
 #
 # Job for linting Habitat shell program components
 #
-    - sudo: false
+    - sudo: true
+      before_script:
+        - ./support/ci/install_shellcheck.sh
```

Comment:
Personally, I'd _rather_ us just use whatever version is installed by default on travis than add extra logic to manage the install (which we also have to keep in sync across our several repos which run shellcheck). I understand the downside is that when travis upgrades, we'll have shellcheck failures at an unexpected moment, but when that happens, we can either:
- Make the fixes if there aren't too many
- Add top-level suppressions in test/shellcheck.sh while we make the fixes
- Temporarily disable shellcheck while we make the fixes

Since shellcheck doesn't update all that often, this may not even come up before we're fully off travis. Additionally, since travis doesn't seem to be aggressive about tracking the latest version of shellcheck (0.46 is from Mar 26, 2017), we could probably put in fixes for newer versions *before* travis upgrades and never need to see any disruption at all. I'd be happy to drive that work.

I worry that if we go the route of having a pinned version, we'll never touch it and never get any benefits of newer versions.

Reply:
I added this in because I installed the latest shellcheck locally, 0.5.0, and then got shellcheck failures when running in Travis because it was using a different version.  Like our other ci scripts for installing zmq, libarchive, libsodium, it felt better to me to install a known version, rather than just trusting whatever version happened to be installed.

Reply:
That's interesting to me that there were things that were failures in 0.4.6 (which runs on travis), but *not* failures in 0.5.0. My general experience had been that newer versions added more checks, so unless there were spurious things in 0.4.6, I'd have expected your local results to be a superset of what the travis run got.